### PR TITLE
docs(api): remove 400 error codes + fix curl queries - fix #184

### PR DESCRIPTION
**Pull request by bsmth**
__Description:__
In cases where API calls succeed but the query fails, we return a `200` with details in the response body for diagnosing the malformed request.

__Additional fixes:__
* multiple minor typos
* fixed CURL examples

### CODE REVIEW

File: docs/reference/api/rest.md [hunk 1]
```diff
@@ -165,7 +166,6 @@ Considering the query:
 curl \
   -F schema='[{"name":"userId", "type": "STRING"},{"name":"movieId", "type":"STRING"}]' \
   -F data=@ratings.csv \
-  --data-urlencode "overwrite=true" \
```
Comment (bsmth) [inline docs/reference/api/rest.md:168]: Side note: this fails for me using __curl 7.64.1 (x86_64-apple-darwin20.0)__:

```
Warning: You can only select one HTTP request method!
```

But this works fine:

```
curl -F data=@myfile.csv http://localhost:9000/imp\?overwrite\=true
```

File: docs/reference/api/rest.md [hunk 2]
```diff
@@ -268,68 +267,34 @@ The HTTP status code will be set to `200` and the response will be:
...
 The parameters must be URL encoded.
 
-### Success response
+### Examples
 
 Considering the query:
 
 ```shell
-curl -G \
+curl -Gv \
```
Comment (mpsq) [inline docs/reference/api/rest.md:291]: `v` is for `verbose` if I am correct, it is not a bad idea having it but we should be consistent, if we introduce `verbose` then we need to do it for all queries (or not, it is up to you, what you think makes more sense to users :) )

**Review by mpsq — CHANGES_REQUESTED** [review]
:tada: congrats on your first PR!
It looks good, just a tiny remark regarding parameters passed to `curl`.

File: docs/reference/api/rest.md [hunk 3]
```diff
@@ -268,68 +267,34 @@ The HTTP status code will be set to `200` and the response will be:
...
   --data-urlencode "query=select AccidentIndex2, Date, Time from 'Accidents0514.csv'" \
   --data-urlencode "limit=5" \
-  http://localhost:9000/exp
+  http://localhost:9000/exec
```
Comment (mpsq) [inline docs/reference/api/rest.md:294]:
```suggestion
  http://localhost:9000/exp
```

This should be `exp`, this example is in the Export data section.

File: docs/reference/api/rest.md [hunk 4]
```diff
@@ -355,37 +320,38 @@ The HTTP status code will be set to `200` and the response will be:
...
 
 Considering the query:
 
 ```shell
 curl -G \
-  --data-urlencode "query=select AccidentIndex2, Date, Time from 'Accidents0514.csv'" \
-  --data-urlencode "limit=5" \
-  http://localhost:9000/exp
+  --data-urlencode "query=SELECTT * FROM table;" \
+  http://localhost:9000/exec
```
Comment (mpsq) [inline docs/reference/api/rest.md:346]:
```suggestion
  http://localhost:9000/exp
```

Comment (bsmth) [reply docs/reference/api/rest.md:346]: Could be either here, right?

File: docs/reference/api/rest.md (hunk 3 above)
Comment (bsmth) [reply docs/reference/api/rest.md:294]: Right 👍

File: docs/reference/api/rest.md (hunk 2 above)
Comment (bsmth) [reply docs/reference/api/rest.md]: removed `-v` flag for simplicity

**Review by mpsq — APPROVED** [review]
LGTM - Nice work

**Pull request merged**